pkg/admission/admission_controller.go

```diff
@@ -115,32 +115,42 @@ func validatePolicies(policies []v1alpha1.LifecyclePolicy, fldPath *field.Path)
 	exitCodes := map[int32]struct{}{}
 
 	for _, policy := range policies {
-		if policy.Event != "" && policy.ExitCode != nil {
+		if len(policy.Event) != 0 && policy.ExitCode != nil {
 			err = multierror.Append(err, fmt.Errorf("must not specify event and exitCode simultaneously"))
 			break
 		}
 
-		if policy.Event == "" && policy.ExitCode == nil {
+		if len(policy.Event) == 0 && policy.ExitCode == nil {
 			err = multierror.Append(err, fmt.Errorf("either event and exitCode should be specified"))
 			break
 		}
 
-		if policy.Event != "" {
-			if allow, ok := policyEventMap[policy.Event]; !ok || !allow {
-				err = multierror.Append(err, field.Invalid(fldPath, policy.Event, fmt.Sprintf("invalid policy event")))
-				break
-			}
-
-			if allow, ok := policyActionMap[policy.Action]; !ok || !allow {
-				err = multierror.Append(err, field.Invalid(fldPath, policy.Action, fmt.Sprintf("invalid policy action")))
-				break
+		if len(policy.Event) != 0 {
+			bFlag := false
+			for _, event := range policy.Event {
+				if allow, ok := policyEventMap[event]; !ok || !allow {
+					err = multierror.Append(err, field.Invalid(fldPath, event, fmt.Sprintf("invalid policy event")))
+					bFlag = true
+					break
+				}
+
+				if allow, ok := policyActionMap[policy.Action]; !ok || !allow {
+					err = multierror.Append(err, field.Invalid(fldPath, policy.Action, fmt.Sprintf("invalid policy action")))
+					bFlag = true
+					break
+				}
+				if _, found := policyEvents[event]; found {
+					err = multierror.Append(err, fmt.Errorf("duplicate event %v", event))
```

Comment:
that's ok if duplicated in the same array.

Reply:
Code updated to ignore duplicates in the same array

Reply:
The duplicated events are already been removed in codes of:
```
policyEventsList := getEventlist(policy)
```
?

Reply:
If there are duplicate events within **one policy** it will be removed in the codes of:
  ` golicyEventList := getEventlist(policy)`

But if there are duplicate events **across different policy** then
  Error message will be thrown `("duplicate events %v",event)`

Reply:
ok, then we may need more clear error message for that since the first case is being handled automatically now.

Reply:
can  the error message be updated  as ` ("duplicate events %v across different policy", event)` ?

Reply:
That would be good thanks.